import.go

```diff
@@ -7,97 +7,89 @@ import (
 	"io"
 	"os"
 	"regexp"
+	"text/template"
 
 	errors "github.com/charter-se/structured/errors"
 	"github.com/spf13/viper"
 )
 
-type TemplateFunc func([]byte) ([]byte, error)
-
-//YamlFile stores raw file bytes and Viper struct
-type YamlSection struct {
-	Bytes []byte
-	Viper *viper.Viper
-}
-
-//Import reads data from a single YAML file and adds its data to this *Yp instance
-func (yp *Yp) ImportFile(s string) error {
-	r, err := os.Open(s)
-	if err != nil {
-		return err
-	}
-	yf, err := importYaml(bufio.NewReader(r))
+//Import takes a location identifier (URI, file path, etc..) and an io.Reader
+//imported data is added to the yamlPack instance
+func (yp *Yp) Import(s string, r io.Reader) error {
+	yf, err := importRawSections(r)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "importRawSections failed in import")
 	}
 	yp.Lock()
 	defer func() {
 		yp.Unlock()
 	}()
 	yp.Files[s] = yf
-	return nil
+	yp.applyNullTemplate(s)
+	return yp.YamlParse(s)
 }
 
-func (yp *Yp) ImportFileWithTemplateFunc(s string, tmplFunc TemplateFunc) error {
+//Import reads data from a single YAML file and adds its data to this *Yp instance
+func (yp *Yp) ImportFile(s string) error {
 	r, err := os.Open(s)
 	if err != nil {
 		return err
 	}
-	yf, err := importYamlWithTemplateFunc(bufio.NewReader(r), tmplFunc)
-	if err != nil {
-		return err
-	}
-	yp.Lock()
-	defer func() {
-		yp.Unlock()
-	}()
-	yp.Files[s] = yf
-	return nil
+	defer r.Close()
+	return yp.Import(s, bufio.NewReader(r))
 }
 
-//Import takes a location identifier (URI, file path, etc..) and an io.Reader
-//imported data is added to the yamlPack instance
-func (yp *Yp) Import(s string, r io.Reader) error {
-	yf, err := importYaml(r)
-	if err != nil {
-		return err
+func (yp *Yp) YamlParse(name string) error {
+	if _, ok := yp.Files[name]; !ok {
+		return errors.WithFields(errors.Fields{
+			"Name": name,
+		}).New("File not imported")
+	}
+	for k, v := range yp.Files {
+		for _, section := range v {
+			//add viper
+			vp := viper.New()
+			vp.SetConfigType("yaml")
+			if err := vp.ReadConfig(bytes.NewBuffer(section.Bytes)); err != nil {
+				fmt.Printf("---\n%v\n", string(section.Bytes))
```

Comment:
Is this a debug statement? Can we control if it is output?

Reply:
Looks like this is done in several areas.

Reply:
It formats the section on error, but it is not controllable. The "structured" package needs to be updated so this can be conveyed without making it a garbled mess.
We can do that in this PR if you suspect its warranted.

Reply:
A separate ticket is fine but we should probably settle on error handling and logging sooner rather than later, or we end with a lot of horizontal cleanup tasks like in Flagship bash :)

Reply:
Agreed.
Can be supported here https://github.com/charter-oss/structured/issues/7.